Lexplorer/Pages/NFTCollection.razor

```diff
@@ -0,0 +1,174 @@
+                                MiddleCount="1" 
+                                ShowLastButton="true" 
+                                Selected="@goToPage"
+                                Count="@goToNextPage"
+                                SelectedChanged="@((int page) => goToPage = page)"/>
+                        </MudToolBar>
+                        </MudPaper>
+                    </MudItem>
+        @if (collectionNFTSlots != null && collectionNFTSlots.Count > 0)
+        {
+            @foreach (var slot in collectionNFTSlots!)
+            {
+                var metaData = GetMetadata(slot.nftID);
+                <MudItem xs="12" sm="4" md="4" lg="2">
+                    <MudCard Style="height:100%;position:relative" Class="ma-1 border-solid">
+                           <MudCardMedia Image="@NftMetadataService.MakeIPFSLink(metaData?.image)" Style="object-fit:contain" />
+                           <MudCardContent>
+                               <MudText Typo="Typo.h6">@metaData?.name</MudText>
+                               <MudText Typo="Typo.body2">@metaData?.description</MudText>
+                           </MudCardContent>
+                           <MudCardActions>
+                               <MudButton Style="position:absolute;bottom:0;" Variant="Variant.Text" Color="Color.Primary" Link=@LinkHelper.GetObjectLinkAddress(slot)?.Item1>Go to NFT</MudButton>
+                           </MudCardActions>
+                    </MudCard>
+                </MudItem>
+            }
+        }
+        else
+        {
+            <MudText class="pa-3">No NFTs</MudText>
+        }
+        <MudHidden Breakpoint="Breakpoint.MdAndDown" Invert="true">
+            <MudPaper Width="100%" Class="mx-4">
+                <MudToolBar Class="justify-center">
+                    <MudPagination Size="Size.Small" ShowFirstButton="true" 
+                    BoundaryCount="1" 
+                    MiddleCount="1" 
+                    ShowLastButton="true" 
+                    Selected="@goToPage"
+                    Count="@goToNextPage"
+                    SelectedChanged="@((int page) => goToPage = page)"/>
+                 </MudToolBar>
+           </MudPaper>
+       </MudHidden>
+       <MudDivider DividerType="DividerType.Middle" Class="my-6"/>
+    </MudGrid>
+</MudContainer>
+
+@code {
+    [Parameter]
+    public string? tokenAddress { get; set; }
+
+    [Parameter]
+    [SupplyParameterFromQuery]
+    public string pageNumber { get; set; } = "1";
+
+    public int goToPage
+    {
+        get
+        {
+            return int.TryParse(pageNumber, out int np) ? np : 1;
+        }
+        set
+        {
+            pageNumber = value.ToString(); //set immediately for MudPagination to not switch forth and back
+            string URL = $"/collections/{tokenAddress}?pageNumber={value}";
+            NavigationManager.NavigateTo(URL);
+        }
+    }
+
+    public int goToNextPage
```

Comment:
This doesn't really add up - goToNextPage implies that it's going somewhere, but you're simply returning what would be the next page. And then you're using this to feed MudPagination.count? So this looks like you're telling MudPagination there's always one more page? I'm surprised this even works, (it does! even though it feels a bit strange when using it), it will never end even if you navigated past the end.

My suggestion would be:
* rename to calculatePageCount
* maybe even re-use the getter of gotoPage instead of TryParse etc.
* do *not* return +1 if collectionNFTSlots < pageSize

Reply:
attempted fix in [ef5e1e5](https://github.com/fudgebucket27/Lexplorer/pull/182/commits/ef5e1e545f2b6d8e0c869aff02c328612c0b75ed)

Reply:
That's a whole lot of code. Why not something simple like:
`
return goToPage + ((collectionNFTSlots?.Count < pageSize) ? 0 : 1);
`

Reply:
i am not a ternary expert as it appears.... fixed as suggested in [abe7fd8](https://github.com/fudgebucket27/Lexplorer/pull/182/commits/abe7fd842eaf4fe21deb20f7974e35fda12bada2)